src/core/crypto.c

```diff
@@ -1720,6 +1720,11 @@ QuicCryptoCustomCertValidationComplete(
             QuicCryptoGetConnection(Crypto),
             "Custom cert validation succeeded");
         QuicCryptoProcessDataComplete(Crypto, Crypto->PendingValidationBufferLength);
+
+        if (QuicRecvBufferHasUnreadData(&Crypto->RecvBuffer)) {
+            // more data was received while waiting for user to perform the validation
+            QuicCryptoProcessData(Crypto, FALSE);
+        }
```

Comment:
It feels more appropriate to keep it separate and not have the logic mutually recursive (QuicCryptoProcessData would call QuicCryptoProcessDataComplete and vice versa).

The additional data are potentially present only when async validations are performed, for synchronous processing, there is never any further additional data since we process everything in single call.